### PR TITLE
Mobifies rockworms

### PR DESCRIPTION
**Pull request by CheffieGithub**
## About the PR 
Mobifies rockworms, lets them throw up ores without being dead so miners have incentive to not kill them all.
Combines fermids and rockworms into mining_critter.dm.

## Why's this needed? 
Object bad. pet worm cute??

## Changelog 

```changelog
(u)Cheffie
(+)Rockworms can now be tamed to not eat ores and can vomit up ores while alive. make a friend today!
Once tamed pet them to switch them between seeking ores and not seeking ores.
```

### CODE REVIEW

File: code/mob/living/critter/mining_critter.dm [hunk 1]
```diff
@@ -153,3 +153,71 @@
...
+	icon_state_dead = "rockworm-dead"
+	hand_count = 1
+	can_lie = FALSE
+	can_throw = FALSE
+	can_grab = FALSE
+	can_disarm = FALSE
+	health_brute = 40
+	health_brute_vuln = 1
+	health_burn = 40
+	health_brute_vuln = 0.1
```
Comment (Vaskritaya) [inline code/mob/living/critter/mining_critter.dm:175]: I think you meant for this to be burn vuln instead

File: code/mob/living/critter/ai/rockworm.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	src.subtasks -= /datum/aiTask/succeedable/critter/eat
+	add_task(holder.get_instance(/datum/aiTask/succeedable/critter/eat/worm, list(src.holder)))
+
+/datum/aiTask/succeedable/critter/eat/worm/on_tick()
+	if(!has_started)
+		var/mob/living/critter/rockworm/C = holder.owner
+		var/obj/item/reagent_containers/food/snacks/T = holder.target
+		if(C && T && BOUNDS_DIST(holder.owner, holder.target) == 0)
+			holder.owner.set_dir(get_dir(holder.owner, holder.target))
+			qdel(T) // Eat code is AAAAAAAAAAAAAAA make it actually eat later
```
[the remark this answers is not in the document]
Comment (CheffieGithub) [reply code/mob/living/critter/ai/rockworm.dm:28]: I dunno how to make it just eat the ores

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Okay I tested this a bit and noticed the following:
- Since they're now living critters they can and will immediately wander off into space and drift off into the void, probably best to either make them not path onto space tiles or give them space traversal.
- You can only tame them with rocks, they will accept any other ore but instead of being tamed they will get angry, this seems backwards. Probably just let them be tamed with ore too.
- Tameable rockworms are very cute.

File: code/mob/living/critter/mining_critter.dm [hunk 3]
```diff
@@ -153,3 +160,108 @@
...
+		AddComponent(/datum/component/consume/can_eat_raw_materials, FALSE)
+
+	is_spacefaring()
+		return TRUE
+
+	on_pet(mob/user)
+		if (..())
+			return 1
+		if (src.tamed && src.ai?.enabled)
+			if (src.ai_type == /datum/aiHolder/rockworm)
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/mining_critter.dm:205]:
```suggestion
			if (src.seek_ore)
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
image

File: code/mob/living/critter/mining_critter.dm (hunk 3 above)
Comment (CheffieGithub) [reply code/mob/living/critter/mining_critter.dm:205]: oh yeah I forgor that

**Pull request merged**